core/src/main/java/org/mobicents/ha/javax/sip/MultiNetworkLoadBalancerHeartBeatingServiceImpl.java

```diff
@@ -452,6 +457,37 @@ protected void updateConnectorsAsSIPNode(SipLoadBalancer loadBalancer) {
 					}
 				}
 				
+				//smpp
+				if(smppPortString == null && smppSslPortString == null) {
+					logger.logWarning("SMPP or secure SMPP port couldn't be retrieved from System properties, trying with JMX");
+					Integer smppPort = null;
+					Boolean smppBound = false;
+					Integer smppSslPort = null;
+					Boolean smppSslBound = false;
+
+					MBeanServer mBeanServer = ManagementFactory.getPlatformMBeanServer();
+					try {
+		                String socketBindingGroup = sipStackProperties.getProperty(SOCKET_BINDING_GROUP, "standard-sockets");
+						ObjectName smpp = new ObjectName("jboss.as:socket-binding-group=" + socketBindingGroup + ",socket-binding=smpp");
+						smppPort = (Integer) mBeanServer.getAttribute(smpp, "boundPort");
+						smppBound = (Boolean) mBeanServer.getAttribute(smpp, "bound");
+
+						ObjectName smppSsl = new ObjectName("jboss.as:socket-binding-group=" + socketBindingGroup + ",socket-binding=smppSsl");
+						smppSslPort = (Integer) mBeanServer.getAttribute(smppSsl, "boundPort");
+						smppSslBound = (Boolean) mBeanServer.getAttribute(smppSsl, "bound");
+						if(logger.isLoggingEnabled(StackLogger.TRACE_TRACE)) {
+							logger.logTrace("Dound smppPort " + smppPort + " and smppSslPort " + smppSslPort);
+						}
+					} catch (Exception e) {} //Ignore any exceptions
+
+					if(smppBound && smppPort!=null){
+						sipStackProperties.setProperty(LOCAL_SMPP_PORT, String.valueOf(smppPort));
+					} 
+					if (smppSslBound && smppSslPort!=null){
+		                sipStackProperties.setProperty(LOCAL_SMPP_SSL_PORT, String.valueOf(smppSslPort));
+					}
+				}
+				
```

Comment:
We don't create any SMPP socket binding group for SMPP in standalone-sip.xml right ? If no, then this part of the code should not be present.